### PR TITLE
I dunno lol; to discuss

### PR DESCRIPTION
**Pull request by ntfshard**
The main question is xs/src/xsinit.h file. A Clone class doesn't look legit but it looks like doesn't needed in dtor(tried - got test fails) but operator= seems to be lost to prevent memory leak

Changes in xs/src/libslic3r/PrintConfig.cpp are obvious(removed dups and trailing spaces)

I wish I had build a gui version but it seems needed some special wxwidgets version

Could you please double check the Clone class

### CODE REVIEW

File: xs/src/xsinit.h [hunk 1]
```diff
@@ -118,6 +120,7 @@ class Clone {
     Clone(const T& t) : val(new T(t)) {}
     operator T*() const { return val; }
     static const char* CLASS() { return ClassTraits<T>::name; }
+    Clone& operator=(const T& t) { delete val; val = new T(t); return *this; }
```
Comment (bubnikv) [inline xs/src/xsinit.h]: This new copy operator is not going to be doing anything. The two wrappers are there to assist Perl to manage the C++ instances by its internal reference counting, so the life of the wrapper is:

1) Created by the XS module
2) Referenced by the Perl, possibly passed back to the C++ code by the pointer of the embedded class.
3) Destructed through Perl's own reference counter.

If there is any cloning being done by the Perl on the XS/C++ instances, then it is when the worker threads are created, see https://perldoc.perl.org/perlmod.html
and the xs/lib/Slic3r/XS.pm and search for sub CLONE_SKIP and  sub clone,
and also Slic3r.pm, sub thread_cleanup where most of the XS/C++ instances are just abandoned by the worker threads.

The whole concept of the Perl worker threads working with the XS/C++ instances, the worker threads being killed hard, and abandoning the XS/C++ data will certainly lead to memory leaks, and we at Prusa Research are now working on rewriting the complete backend into C++, so there will be no Perl worker threads running in Slic3r PE 1.41.

**Comment by lordofhyphens** [comment]
Please don't mix code cleanup/refactor with other changes. @bubnikv is also correct about the lifetime of the wrappers. 

Closing.

**Comment by ntfshard** [comment]
I see, thank you.
Also I tested values in assigment operator(aka operator= ) and all of them are NULL, so it looks legit to use default
